docs/environment-and-software/user-environment/spack/spack.md

```diff
@@ -119,4 +119,461 @@ You could also limit the scope of this specific projection to **parallel-netcdf*
+    [^]  sxofwmo      ^parmetis@4.0.3%oneapi@2023.2.1~gdb~int64~ipo+shared build_system=cmake build_type=Release generator=make patches=4f89253,50ed208,704b84f arch=linux-sles15-x86_64_v3
+    [^]  lmsy2vj      ^python@3.10.12%gcc@7.5.0+bz2+crypt+ctypes+dbm~debug+libxml2+lzma~nis~optimizations+pic+pyexpat+pythoncmd+readline+shared+sqlite3+ssl~tkinter+uuid+zlib build_system=generic patches=0d98e93,7d40923,f2fd060 arch=linux-sles15-x86_64_v3
+    [^]  adam562          ^bzip2@1.0.8%gcc@7.5.0~debug~pic+shared build_system=generic arch=linux-sles15-x86_64_v3
+    [^]  ioxwsfz          ^expat@2.5.0%gcc@7.5.0+libbsd build_system=autotools arch=linux-sles15-x86_64_v3
+    [^]  fb4m5js              ^libbsd@0.11.7%gcc@7.5.0 build_system=autotools arch=linux-sles15-x86_64_v3
+    [^]  huw6hi3                  ^libmd@1.0.4%gcc@7.5.0 build_system=autotools arch=linux-sles15-x86_64_v3
+    [^]  wsmcahk          ^gdbm@1.23%gcc@7.5.0 build_system=autotools arch=linux-sles15-x86_64_v3
+    [^]  al2lizj          ^gettext@0.21.1%gcc@7.5.0+bzip2+curses+git~libunistring+libxml2+tar+xz build_system=autotools arch=linux-sles15-x86_64_v3
+    [^]  k4pmi3f              ^libiconv@1.17%gcc@7.5.0 build_system=autotools libs=shared,static arch=linux-sles15-x86_64_v3
+    [^]  p2qilsy              ^libxml2@2.10.3%gcc@7.5.0~python build_system=autotools arch=linux-sles15-x86_64_v3
+    [e]  c4brufk              ^tar@1.34%gcc@7.5.0 build_system=autotools zip=pigz arch=linux-sles15-x86_64_v3
+    [^]  as5rojq          ^libffi@3.3%gcc@7.5.0 build_system=autotools patches=26f26c6 arch=linux-sles15-x86_64_v3
+    [^]  u5teb4s          ^libxcrypt@4.4.35%gcc@7.5.0~obsolete_api build_system=autotools patches=4885da3 arch=linux-sles15-x86_64_v3
+    [^]  w4gepfm          ^ncurses@6.4%gcc@7.5.0~symlinks+termlib abi=none build_system=autotools arch=linux-sles15-x86_64_v3
+    [e]  ohtkism          ^openssl@1.1.1l%gcc@7.5.0~docs+shared build_system=generic certs=system arch=linux-sles15-x86_64_v3
+    [^]  npfqu2h          ^readline@8.2%gcc@7.5.0 build_system=autotools patches=bbf97f1 arch=linux-sles15-x86_64_v3
+    [^]  a5yrh7q          ^sqlite@3.42.0%gcc@7.5.0+column_metadata+dynamic_extensions+fts~functions+rtree build_system=autotools arch=linux-sles15-x86_64_v3
+    [^]  rhcskli          ^util-linux-uuid@2.38.1%gcc@7.5.0 build_system=autotools arch=linux-sles15-x86_64_v3
+    [^]  fvguvlm          ^xz@5.4.1%gcc@7.5.0~pic build_system=autotools libs=shared,static arch=linux-sles15-x86_64_v3
+     -   nwww45j      ^superlu-dist@8.2.1%gcc@7.5.0~cuda~int64~ipo+openmp+parmetis~rocm+shared build_system=cmake build_type=Release generator=make arch=linux-sles15-x86_64_v3
+    [^]  g42iifh      ^zlib@1.2.13%gcc@7.5.0+optimize+pic+shared build_system=makefile arch=linux-sles15-x86_64_v3
+    ```
+
+
+
+To fix this issue we need to be very explicit with the concretization by requiring the additional dependencies to be compiled with the same compiler.
```

Comment:
Why the switch here from OneAPI above to GCC 12 here?

Also, I think it would be helpful explaining *why* hypre and superlu-dist are special amongst the many other dependencies. I assume it is because they themselves depend on MPI, but in any case the average user probably won't know.

Reply:
(And if it's just to show variety in compiler choice, still, I think you should have the CPU-case show up first below or use the GPU case above. The apparently change in compilers distracts from the main point IMO.)

Reply:
Resolved.